File: frontend/lib/des_ede3.dart
import 'dart:convert';
import 'dart:typed_data';
import 'package:dart_des/dart_des.dart';

class DesEde3 {
  static String encrypt(String plainText, String key) {
    List<int> encodeKey = utf8.encode(key).getRange(0, 8).toList();
    DES cipher = DES(key: encodeKey, mode: DESMode.ECB);

    plainText += '\x00' * (8 - plainText.length % 8);
    List<int> encrypted = cipher.encrypt(ascii.encode(plainText));
    String cipherBase64 = base64Encode(encrypted);

    return utf8.decode(cipherBase64.codeUnits);
  }

  static String decrypt(String cipherText, String key) {
    List<int> decodeKey = key.substring(0, 8).codeUnits;
    DES cipher = DES(key: decodeKey, mode: DESMode.ECB);
    Uint8List plainTextBase64 = base64Decode(cipherText);
    List<int> s = cipher.decrypt(plainTextBase64);

    return utf8.decode(s);
  }
}

File: server/lib/des_ede3.py
from Crypto.Cipher import DES3
import base64

class DesEde3:
    def __init__(self) -> None:
        pass

    # debug
    def byte_to_arr(self, byte):
        return list(bytearray(byte))

    def encrypt(self, plain_text: str, key: str) -> str:
        encoded_key = self.encode_key(key)
        crypter = DES3.new(encoded_key, DES3.MODE_ECB)

        plain_text += '\x00' * (8 - len(plain_text) % 8)
        encrypted: bytes = crypter.encrypt(plain_text.encode('ascii'))
        cipter_base64: bytes = base64.standard_b64encode(encrypted)

        return cipter_base64.decode("utf-8")

    def decrypt(self, cipher_text: str, key: str) -> str:
        encoded_key = self.encode_key(key)
        crypter = DES3.new(encoded_key, DES3.MODE_ECB)
        
        utf8 = cipher_text.encode('utf-8')
        cipher_base64 = base64.standard_b64decode(utf8)
        plain_text = crypter.decrypt(cipher_base64)
        
        return plain_text.decode('utf-8', errors='ignore')


    def encode_key(self, key: str):
        return key.encode('utf-8')[0:16]